smaily_for_drupal/src/Form/SmailyForDrupalForm.php

```diff
@@ -0,0 +1,126 @@
+<?php
+
+namespace Drupal\smaily_for_drupal\Form;
+
+use Drupal\Core\Form\FormBase;
+use Drupal\Core\Form\FormStateInterface;
+
+/**
+ * Smaily newsletter signup form.
+ */
+class SmailyForDrupalForm extends FormBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getFormId() {
+    return 'smaily_for_drupal_form';
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function buildForm(array $form, FormStateInterface $form_state) {
+    $config = $this->config('smaily_for_drupal.adminsettings');
+
+    $form['name'] = [
+      '#type' => 'textfield',
+      '#title' => $this->t('Name:'),
+      '#attributes' => ['placeholder' => $this->t('Name')],
+    ];
+
+    $form['email'] = [
+      '#type' => 'email',
+      '#title' => $this->t('Email:'),
+      '#attributes' => ['placeholder' => $this->t('Email')],
+      '#required' => TRUE,
+    ];
+
+    $form['subscribe'] = [
+      '#type' => 'submit',
+      '#value' => $config->get('smaily_button_title', ''),
+      '#default_value' => $this->t('Subscribe to newsletter'),
+    ];
+    return $form;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function validateForm(array &$form, FormStateInterface $form_state) {
+    // #type => 'email' does validation.
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function submitForm(array &$form, FormStateInterface $form_state) {
+    $config = $this->config('smaily_for_drupal.adminsettings');
+
+    $username = $config->get('smaily_username');
+    $password = $config->get('smaily_password');
+    $domain = $config->get('smaily_domain');
+
+    $query_data = [
+      'remote' => 1,
+      'email' => $form_state->getValue('email'),
+    ];
+
+    $name = $form_state->getValue('name');
+    if (!empty($name)) {
+      $query_data['name'] = $name;
+    }
+
+    $autoresponder = $config->get('smaily_autoresponder');
+    if (!empty($autoresponder)) {
+      $query_data['autoresponder'] = $autoresponder;
+    }
+
+    $ch = curl_init();
+    curl_setopt($ch, CURLOPT_URL, 'https://' . $domain . '.sendsmaily.net/api/opt-in/');
```

Comment:
Use API endpoint `/api/autoresponder.php`. Please take a look at the [API documentation](http://help.smaily.com/en/support/solutions/articles/16000021289-autoresponder).

Reply:
What's the difference of this and `/api/opt-in/`?

Reply:
There is none. But why then configure Smaily API credentials if you are not going to use them? :)

Reply:
I reworked it for autoresponder.php. Autoresponder.php asks for some authorization while /opt-in/ doesn't, yes? And I already went through the trouble of getting API credentials from config, so might as well cURL to autoresponder.php
Is the current solution OK or did I misunderstand you?

Reply:
Yes, that is mighty fine. Thanks!

`/api/autoresponder.php` uses authentication and it is more meant for background requests.  `/api/opt-in/` is more meant for embedded opt-in forms that directly submit form's contents to Smaily. We are trying to make this separation at the moment with all of our integrations.